### PR TITLE
improvement/E_CLASSROOM-244-FE/BE-Profile-picture-upload

### PR DESCRIPTION
**Pull request by CaneteJoash**
### Issue Link
https://framgiaph.backlog.com/view/E_CLASSROOM-244

### Definition of Done
- [x] User can upload a picture
- [x] A maximum size of 1MB only
- [x] Should only accept files of type picture (png, jpg, ...)

### Commands to Run
N/A

### Related PR:
- https://github.com/framgia/sph-classroom-els-be/pull/46

#### Main note
- N/A
#### Pages Affected
- http://localhost:3003/profile

### Scenarios/ Test Cases
- [x] User can upload a picture
- [x] A maximum size of 1MB only
- [x] Should only accept files of type picture (png, jpg, ...)
#### User Interface
N/A
#### User Experience 
N/A`
#### Functionality
N/A
### Screenshots

image
image

### test view points 

- [x] Try upload image that has size of 2mb plus
- [x] Try upload text pad like notepad
- [x] Try upload docs 

image
image

### CODE REVIEW

File: src/pages/student/profile/ProfileDetail/index.js [hunk 1]
```diff
@@ -30,18 +29,33 @@ const ProfileDetail = () => {
...
-    } 
-    catch (error) {
+      await ProfileEditApi.uploadImage(data);
+      setSuccessMessage('Upload Successful, Please refresh the page');
+      setStatus(true);
+    } catch (error) {
       console.log(error.response);
+      if (error?.response?.data?.errors) {
+        setError(error?.response?.data?.errors);}
+      setSubmitStatus(false);
```
Comment (karlolee50) [inline src/pages/student/profile/ProfileDetail/index.js]: use the toaster and remove console log if not used

File: src/pages/student/profile/ProfileDetail/index.js [hunk 2]
```diff
@@ -30,18 +29,33 @@ const ProfileDetail = () => {
...
-    console.log(image[0]);
+  const handleOnSubmit = async (image) => {
+    let data = new FormData();
+    data.append('image', image.image);
     try {
-      await ProfileEditApi.uploadImage({ image });
-    } 
-    catch (error) {
+      await ProfileEditApi.uploadImage(data);
+      setSuccessMessage('Upload Successful, Please refresh the page');
```
Comment (karlolee50) [inline src/pages/student/profile/ProfileDetail/index.js:47]: use toaster

File: src/pages/student/profile/ProfileDetail/index.js [hunk 3]
```diff
@@ -30,18 +29,33 @@ const ProfileDetail = () => {
...
+      if (error?.response?.data?.errors) {
+        setError(error?.response?.data?.errors);}
+      setSubmitStatus(false);
     }
   };
 
   useEffect(() => {
+    StudentApi.getDetails(loggedInUserId).then(({ data }) => {
+      setAvatar(data.avatar);
+      console.log(data.avatar);
```
Comment (karlolee50) [inline src/pages/student/profile/ProfileDetail/index.js]: remove console log if not used

**Comment by Teraaay** [comment]
### Working
- [x] Able to upload a photo and save it in the database.
- [x] File upload only accepts a file of type: jpeg, png, jpg, gif, svg.

### Comment/s
- [x] The profile picture that was uploaded is not displaying properly.

Please refer to the screenshot below.
     
### Screenshot/s
image

------
- Total Test Time: 0.25 hr

**Comment by Teraaay** [comment]
### Approved PR #85 
- [x] Able to upload a photo and save it in the database.
- [x] File upload only accepts a file of type: jpeg, png, jpg, gif, svg.
- [x] Able to display the uploaded image.

### Screenshot/s | Screen Record/s

UPLOAD PROFILE PIC

----
- Total Test time: 0.5 hr
- Total Iterations: 1
- Total Comments: 2

**Review by karlolee50 — APPROVED** [review]

**Pull request merged**